File: src/main/java/com/windea/demo/csntportal/service/DevelopmentColumnService.java
package com.windea.demo.csntportal.service;

import com.windea.demo.csntportal.domain.entity.DevelopmentColumn;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * 专业发展专栏的服务接口。
 */
public interface DevelopmentColumnService {
	DevelopmentColumn create(DevelopmentColumn column);

	void delete(Integer id);

	DevelopmentColumn update(DevelopmentColumn column);

	DevelopmentColumn get(Integer id);

	Page<DevelopmentColumn> list(Pageable pageable);

	Page<DevelopmentColumn> searchByTitle(String title, Pageable pageable);
}

File: src/main/mobile/src/app/service/api/development-column.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {DevelopmentColumn} from "../../domain/entity/DevelopmentColumn";
import {Observable} from "rxjs";
import {apiUrl} from "../../../environments/environment.prod";
import {Page} from "../../domain/interface/Page";

/**
 * 专业发展专栏的服务类。
 */
@Injectable({providedIn: "root"})
export class DevelopmentColumnService {
  constructor(private http: HttpClient) {
  }


  create(column: DevelopmentColumn): Observable<DevelopmentColumn> {
    const url = `${apiUrl}/development-column/create`;
    return this.http.post<DevelopmentColumn>(url, column);
  }

  delete(id: number) {
    const url = `${apiUrl}/development-column/${id}`;
    return this.http.delete(url);
  }

  update(column: DevelopmentColumn): Observable<DevelopmentColumn> {
    const url = `${apiUrl}/development-column/${column.id}`;
    return this.http.put<DevelopmentColumn>(url, column);
  }

  get(id: number): Observable<DevelopmentColumn> {
    const url = `${apiUrl}/development-column/${id}`;
    return this.http.get<DevelopmentColumn>(url);
  }

  list(page: number, size: number): Observable<Page<DevelopmentColumn>> {
    const url = `${apiUrl}/development-column/list`;
    const params = {page: page + "", size: size + ""};
    return this.http.get<Page<DevelopmentColumn>>(url, {params: params});
  }

  searchByTitle(title: string, page: number, size: number): Observable<Page<DevelopmentColumn>> {
    const url = `${apiUrl}/development-column/search`;
    const params = {title: title, page: page + "", size: size + ""};
    return this.http.get<Page<DevelopmentColumn>>(url, {params: params});
  }
}
